Import bug in Next.js (v13.4.13)

## Prerequisites

* Version: 0.264.0
* Are you running from source/main: nop
* Are you using a released build: nop
* Operating system: Mac OS
* Bits: -

## Step to reproduce

Import icon in a Next.js v13.4.13 (latest)

### Actual behavior

I see an error when importing the icon.

## Any message or error

```
Attempted import error: 'modularize-import-loader?name=CalendarDays&from=default&as=default&join=./icons/calendar-days!lucide-react' does not contain a default export (imported as 'CalendarDays').
```

--- 

It works well on v0.263.x, I suspect a bug in the new ESM export.

error with 0.264.0
next 13.4.13

- error ./node_modules/lucide-react/dist/cjs/lucide-react.js:2:0
Module not found: Can't resolve '/Users/abdelhameed/DevApps/vercel/my-app-shadcn/node_modules/lucide-react/dist/cjs/icons/chevron-left'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./components/ui/calendar.tsx
./app/page.tsx

next.config:
```
/** @type {import('next').NextConfig} */
const nextConfig = {}

module.exports = nextConfig
```

I am currently facing this issue with Next.js 13.4.13 and lucide-react version 0.264.0 

- error ./node_modules/lucide-react/dist/cjs/lucide-react.js:2:0
Module not found: Can't resolve 'D:\grafbash_fullstack\node_modules\lucide-react\dist\cjs\icons\camera'

My workaround: Downgraded lucide react to version 0.263.1 and it is working like expected.

**Packages version**

`next: "^13.4.13"`
`lucide-ract : "^0.265.0"`

Getting Error while importing Icons

* Icons.tsx
```ts
import { PackageCheck, PackageMinus, type LucideIcon } from 'lucide-react';


export type Icon = LucideIcon


export const Icons = {
    logo: PackageMinus
}
```

* MainNav.tsx
```ts
"use client"
import Link from "next/link";
import { useSelectedLayoutSegment } from "next/navigation"
import { useState } from "react";
import { Icons } from "@/components/Icons";

interface MainNavProps {


}

export function MainNav() {
    const segment = useSelectedLayoutSegment()
    const [mobileMenu, setMobileMenu] = useState<boolean>(false)

    return (
        <div className="flex gap-6 md:gap-10">
            <Link href="/" className="hidden items-center space-x-2 md:flex">
                <Icons.logo />
            </Link>
        </div >
    )
}
```


***Error :*** 
```bash
Module not found: Can't resolve '/Users/sonu/Projects/NextJS/memoir/node_modules/lucide-react/dist/cjs/icons/package-minus'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./components/Icons.tsx
./components/MainNav.tsx
```

Same problem

`next: 13.4.13`
`lucide-react: 0.265.0`

```
- error ./node_modules/.pnpm/lucide-react@0.265.0_react@18.2.0/node_modules/lucide-react/dist/cjs/lucide-react.js:2:0
Module not found: Can't resolve '/Users/zdy/Desktop/Project/Nextjs_Learn/node_modules/.pnpm/lucide-react@0.265.0_react@18.2.0/node_modules/lucide-react/dist/cjs/icons/check'
```

--- a/.github/workflows/lucide-angular.yml
+++ b/.github/workflows/lucide-angular.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-angular/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-font.yml
+++ b/.github/workflows/lucide-font.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - icons/**
+      - tools/build-font/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-preact.yml
+++ b/.github/workflows/lucide-preact.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-preact/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-react-native.yml
+++ b/.github/workflows/lucide-react-native.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-react-native/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-react.yml
+++ b/.github/workflows/lucide-react.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-react/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-solid.yml
+++ b/.github/workflows/lucide-solid.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-solid/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-static.yml
+++ b/.github/workflows/lucide-static.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-static/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-svelte.yml
+++ b/.github/workflows/lucide-svelte.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-svelte/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-vue-next.yml
+++ b/.github/workflows/lucide-vue-next.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-vue-next/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide-vue.yml
+++ b/.github/workflows/lucide-vue.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide-vue/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/.github/workflows/lucide.yml
+++ b/.github/workflows/lucide.yml
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - packages/lucide/**
+      - tools/build-icons/**
       - pnpm-lock.yaml
 
 jobs:

--- a/packages/lucide-react/package.json
+++ b/packages/lucide-react/package.json
@@ -28,7 +28,7 @@
     "build": "pnpm clean && pnpm copy:license && pnpm build:icons && pnpm typecheck && pnpm build:bundles",
     "copy:license": "cp ../../LICENSE ./LICENSE",
     "clean": "rm -rf dist && rm -rf stats && rm -rf ./src/icons/*.ts && rm -f dynamicIconImports.*",
-    "build:icons": "build-icons --output=./src --templateSrc=./scripts/exportTemplate.mjs --renderUniqueKey --withAliases --withDynamicImports --aliasesFileExtension=.ts --iconFileExtension=.ts --exportFileName=index.ts",
+    "build:icons": "build-icons --output=./src --templateSrc=./scripts/exportTemplate.mjs --renderUniqueKey --withAliases --withDynamicImports --separateAliasesFile --aliasesFileExtension=.ts --iconFileExtension=.ts --exportFileName=index.ts",
     "build:types": "node ./scripts/buildTypes.mjs",
     "build:bundles": "rollup -c ./rollup.config.mjs",
     "typecheck": "tsc",

--- a/packages/lucide-react/package.json
+++ b/packages/lucide-react/package.json
@@ -1,7 +1,7 @@
 {
   "name": "lucide-react",
   "description": "A Lucide icon library package for React applications",
-  "version": "0.263.2-beta.0",
+  "version": "0.0.1",
   "license": "ISC",
   "homepage": "https://lucide.dev",
   "bugs": "https://github.com/lucide-icons/lucide/issues",

--- a/packages/lucide-react/rollup.config.mjs
+++ b/packages/lucide-react/rollup.config.mjs
@@ -1,6 +1,9 @@
 import plugins, { replace } from '@lucide/rollup-plugins';
 import pkg from './package.json' assert { type: 'json' };
 import dts from "rollup-plugin-dts";
+import getAliasesEntryNames from './scripts/getAliasesEntryNames.mjs';
+
+const aliasesEntries = await getAliasesEntryNames()
 
 const packageName = 'LucideReact';
 const outputFileName = 'lucide-react';
@@ -26,7 +29,10 @@ const bundles = [
   },
   {
     format: 'esm',
-    inputs,
+    inputs: [
+      ...inputs,
+      ...aliasesEntries
+    ],
     outputDir,
     preserveModules: true,
     aliasesSupport: true,

--- a/packages/lucide-react/scripts/getAliasesEntryNames.mjs
+++ b/packages/lucide-react/scripts/getAliasesEntryNames.mjs
@@ -1,0 +1,15 @@
+import path from 'path';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import getAliases from '@lucide/build-icons/utils/getAliases.mjs';
+
+const ICONS_DIR = path.resolve(process.cwd(), '../../icons');
+
+export default async function getAliasesEntryNames() {
+  const metaJsonFiles = await getAliases(ICONS_DIR);
+
+  const iconWithAliases = Object.values(metaJsonFiles).filter(({ aliases }) => aliases != null);
+
+  const aliases = iconWithAliases.flatMap(({ aliases }) => aliases);
+
+  return aliases.map((alias) => path.join('src/icons', `${alias}.ts`));
+}

--- a/tools/build-icons/building/generateAliasesFile.mjs
+++ b/tools/build-icons/building/generateAliasesFile.mjs
@@ -37,34 +37,27 @@ export default async function generateAliasesFile({
           iconAliases.map(async (alias) => {
             const componentNameAlias = toPascalCase(alias);
             const output = `export { default } from "./${iconName}"`;
-            const location = path.join(
-              iconsDistDirectory,
-              `${componentNameAlias}${iconFileExtension}`
-            );
+            const location = path.join(iconsDistDirectory, `${alias}${iconFileExtension}`);
             await fs.promises.writeFile(location, output, 'utf-8');
-            importString += getImportString(
-              componentNameAlias,
-              componentNameAlias,
-              aliasImportFileExtension
-            );
+            importString += getImportString(componentNameAlias, alias, aliasImportFileExtension);
             importString += getImportString(
               `${componentNameAlias}Icon`,
-              componentNameAlias,
-              aliasImportFileExtension
+              alias,
+              aliasImportFileExtension,
             );
             importString += getImportString(
               `Lucide${componentNameAlias}`,
-              componentNameAlias,
-              aliasImportFileExtension
+              alias,
+              aliasImportFileExtension,
             );
-          })
+          }),
         );
       }
 
       importString += '\n';
 
       appendFile(importString, fileName, outputDirectory);
-    })
+    }),
   );
 
   appendFile('\n', fileName, outputDirectory);

--- a/tools/build-icons/building/generateAliasesFile.mjs
+++ b/tools/build-icons/building/generateAliasesFile.mjs
@@ -12,6 +12,7 @@ export default async function generateAliasesFile({
   iconFileExtension = '.js',
   aliases,
   aliasImportFileExtension,
+  separateAliasesFile = false,
   showLog = true,
 }) {
   const iconsDistDirectory = path.join(outputDirectory, `icons`);
@@ -36,18 +37,30 @@ export default async function generateAliasesFile({
         await Promise.all(
           iconAliases.map(async (alias) => {
             const componentNameAlias = toPascalCase(alias);
-            const output = `export { default } from "./${iconName}"`;
-            const location = path.join(iconsDistDirectory, `${alias}${iconFileExtension}`);
-            await fs.promises.writeFile(location, output, 'utf-8');
-            importString += getImportString(componentNameAlias, alias, aliasImportFileExtension);
+
+            if (separateAliasesFile) {
+              const output = `export { default } from "./${iconName}"`;
+              const location = path.join(iconsDistDirectory, `${alias}${iconFileExtension}`);
+
+              await fs.promises.writeFile(location, output, 'utf-8');
+            }
+
+            const exportFileIcon = separateAliasesFile ? alias : iconName;
+
             importString += getImportString(
-              `${componentNameAlias}Icon`,
-              alias,
+              componentNameAlias,
+              exportFileIcon,
               aliasImportFileExtension,
             );
             importString += getImportString(
+              `${componentNameAlias}Icon`,
+              exportFileIcon,
+              aliasImportFileExtension,
+            );
+
+            importString += getImportString(
               `Lucide${componentNameAlias}`,
-              alias,
+              exportFileIcon,
               aliasImportFileExtension,
             );
           }),

--- a/tools/build-icons/building/generateAliasesFile.mjs
+++ b/tools/build-icons/building/generateAliasesFile.mjs
@@ -1,17 +1,20 @@
 import path from 'path';
+import fs from 'fs';
 import { toPascalCase, resetFile, appendFile } from '../../../scripts/helpers.mjs';
 
 const getImportString = (componentName, iconName, aliasImportFileExtension = '') =>
   `export { default as ${componentName} } from './icons/${iconName}${aliasImportFileExtension}';\n`;
 
-export default function generateAliasesFile({
+export default async function generateAliasesFile({
   iconNodes,
   outputDirectory,
   fileExtension,
+  iconFileExtension = '.js',
   aliases,
   aliasImportFileExtension,
   showLog = true,
 }) {
+  const iconsDistDirectory = path.join(outputDirectory, `icons`);
   const fileName = path.basename(`aliases${fileExtension}`);
   const icons = Object.keys(iconNodes);
 
@@ -19,28 +22,50 @@ export default function generateAliasesFile({
   resetFile(fileName, outputDirectory);
 
   // Generate Import for Icon VNodes
-  icons.forEach((iconName) => {
-    const componentName = toPascalCase(iconName);
-    const iconAliases = aliases[iconName]?.aliases;
+  await Promise.all(
+    icons.map(async (iconName) => {
+      const componentName = toPascalCase(iconName);
+      const iconAliases = aliases[iconName]?.aliases;
 
-    let importString = `// ${componentName} aliases\n`;
+      let importString = `// ${componentName} aliases\n`;
 
-    importString += getImportString(`${componentName}Icon`, iconName, aliasImportFileExtension);
-    importString += getImportString(`Lucide${componentName}`, iconName, aliasImportFileExtension);
+      importString += getImportString(`${componentName}Icon`, iconName, aliasImportFileExtension);
+      importString += getImportString(`Lucide${componentName}`, iconName, aliasImportFileExtension);
 
-    if (iconAliases != null && Array.isArray(iconAliases)) {
-      iconAliases.forEach((alias) => {
-        const componentNameAlias = toPascalCase(alias);
-        importString += getImportString(componentNameAlias, iconName, aliasImportFileExtension);
-        importString += getImportString(`${componentNameAlias}Icon`, iconName, aliasImportFileExtension);
-        importString += getImportString(`Lucide${componentNameAlias}`, iconName, aliasImportFileExtension);
-      });
-    }
+      if (iconAliases != null && Array.isArray(iconAliases)) {
+        await Promise.all(
+          iconAliases.map(async (alias) => {
+            const componentNameAlias = toPascalCase(alias);
+            const output = `export { default } from "./${iconName}"`;
+            const location = path.join(
+              iconsDistDirectory,
+              `${componentNameAlias}${iconFileExtension}`
+            );
+            await fs.promises.writeFile(location, output, 'utf-8');
+            importString += getImportString(
+              componentNameAlias,
+              componentNameAlias,
+              aliasImportFileExtension
+            );
+            importString += getImportString(
+              `${componentNameAlias}Icon`,
+              componentNameAlias,
+              aliasImportFileExtension
+            );
+            importString += getImportString(
+              `Lucide${componentNameAlias}`,
+              componentNameAlias,
+              aliasImportFileExtension
+            );
+          })
+        );
+      }
 
-    importString += '\n';
+      importString += '\n';
 
-    appendFile(importString, fileName, outputDirectory);
-  });
+      appendFile(importString, fileName, outputDirectory);
+    })
+  );
 
   appendFile('\n', fileName, outputDirectory);
 

--- a/tools/build-icons/main.mjs
+++ b/tools/build-icons/main.mjs
@@ -62,9 +62,10 @@ async function buildIcons() {
   if (withAliases) {
     const aliases = await getAliases(ICONS_DIR);
 
-    generateAliasesFile({
+    await generateAliasesFile({
       iconNodes: icons,
       aliases,
+      iconFileExtension,
       outputDirectory: OUTPUT_DIR,
       fileExtension: aliasesFileExtension,
       aliasImportFileExtension,

--- a/tools/build-icons/main.mjs
+++ b/tools/build-icons/main.mjs
@@ -32,6 +32,7 @@ const {
   exportFileName = 'index.js',
   withAliases = false,
   withDynamicImports = false,
+  separateAliasesFile = false,
   aliasesFileExtension = '.js',
   aliasImportFileExtension = '',
   pretty = true,
@@ -69,6 +70,7 @@ async function buildIcons() {
       outputDirectory: OUTPUT_DIR,
       fileExtension: aliasesFileExtension,
       aliasImportFileExtension,
+      separateAliasesFile,
       showLog: !silent,
     });
   }

